web_sdk/web_test_utils/lib/skia_client.dart

```diff
@@ -279,6 +288,40 @@ class SkiaGoldClient {
     }
   }
 
+  List<String> _getMatchingArguments(String testName, int screenshotSize) {
+    // The algorithm to be used when matching images. The available options are:
+    // - "fuzzy": Allows for customizing the thresholds of pixel differences.
+    // - "sobel": Same as "fuzzy" but performs edge detection before performing
+    //            a fuzzy match.
+    const String algorithm = 'fuzzy';
+
+    // The number of pixels in this image that are allowed to differ from the
+    // baseline. It's okay for this to be a slightly high number like 10% of the
+    // image size because those wrong pixels are constrained by
+    // `pixelDeltaThreshold` below.
+    final int maxDifferentPixels = (screenshotSize * kMaxDifferentPixelsRate).toInt();
+
+    // The maximum acceptable difference in RGB channels of each pixel.
+    //
+    // ```
+    // abs(r(image) - r(golden)) + abs(g(image) - g(golden)) + abs(b(image) - b(golden)) <= pixelDeltaThreshold
+    // ```
+    final String pixelDeltaThreshold;
+    if (testName.startsWith('canvaskit_')) {
+      pixelDeltaThreshold = '21';
```

Comment:
We currently have zero diff tolerance for canvaskit screenshots? Why 21?

Reply:
I picked those numbers based on the false negatives that I've been seeing over the last few days. In canvaskit's case, there are several images that had a difference of (7,7,7) and (5,5,5) from their baselines.

Reply:
Interesting. Do you know why they don't fail today?

Reply:
You mean in flutter/goldens? I'm not sure. I assumed we were applying some kinda threshold but I didn't check.

Reply:
Ah, you're right. We use 0% pixel _count_ threshold, but each pixel is allowed to be 10% different from the golden pixel. This is controllable using `PixelComparison` but few tests use it.